tests/testsearch.py

```diff
@@ -640,6 +640,76 @@ def testFileUpload(self):
+        result = list(q2())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("Query with chain ID results: result length : (%d), ok : (%r)", len(result), ok)
+
+        # Query with file url
+        q3 = StructSimilarityQuery("file_url", "https://files.rcsb.org/view/4HHB.cif", input_structure_id="cif")
+        result = list(q3())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("Query with file url results: result length : (%d), ok : (%r)", len(result), ok)
+
+        # Query with file upload
+        q4 = StructSimilarityQuery("file_upload", self.__4hhbCif, input_structure_id="cif")
+        result = list(q4())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("Query with file upload results: result length : (%d), ok : (%r)", len(result), ok)
+
+        # Query with relaxed operator
+        q5 = StructSimilarityQuery("entry_id", "4HHB", operator="relaxed_shape_match")
+        result = list(q5())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("Query with relaxed operator results: result length : (%d), ok : (%r)", len(result), ok)
+
+        # Query with specifically polymer entity instance search space
+        q6 = StructSimilarityQuery(structure_search_type="entry_id",
+                                   value="4HHB",
+                                   input_structure_type="chain_id",
+                                   input_structure_id="B",
+                                   operator="relaxed_shape_match",
+                                   target_search_space="polymer_entity_instance")
+        result = list(q6())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("Query with polymer entity instance results: result length : (%d), ok : (%r)", len(result), ok)
+
+        # File upload query using 4HHB Assembly 1 - cif zip file
+        q7 = StructSimilarityQuery("file_upload", self.__4hhbAssembly1, input_structure_id="cif")
+        result = list(q7())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("File upload query using 4HHB Assembly 1 cif zip file results : (%d), ok : (%r)", len(result), ok)
+
+        # File upload query using 4HHB PDB file
+        q8 = StructSimilarityQuery("file_upload", self.__4hhbPdb, input_structure_id="pdb")
+        result = list(q8())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("File upload query using 4HHB PDB file results: result length : (%d), ok : (%r)", len(result), ok)
+
+        # File upload query using 4HHB bcif file
+        q9 = StructSimilarityQuery("file_upload", self.__4hhbBcif, input_structure_id="bcif")
+        result = list(q9())
+        ok = len(result) > 0
+        self.assertTrue(ok)
+        logger.info("File upload query using 4HHB bcif file results: result length : (%d), ok : (%r)", len(result), ok)
+
```

Comment:
I suggest to add a test with `file_url` + `mmCIF` + `relaxed` + chains. 
Another test that should fail is the wrong combination of `file_url` + format (i.e., cif file url + pdb format).